go/types/meta_sequence.go

```diff
@@ -177,26 +335,29 @@ func (ms metaSequence) getCompositeChildSequence(start uint64, length uint64) se
 		isIndexedSequence = true
 	}
 
+	// TODO: This looks strange. The children can only be a meta sequence or one of map/set/list
```

Comment:
yeah. now that this is byte-backed, we could construct new values directly in a kind of streaming way. can you open a bug to do that. that'd probably avoid alot of allocation.

Reply:
https://github.com/attic-labs/noms/issues/3706